### ISSUE TITLE
How to convert qt designer file(.ui) to go code?

### ISSUE THREAD

**Issue by tnajor** [therecipe/qt#845]
It will be easy for normal developer if there is a way from qt designer file(.ui) to go code.

**Comment by therecipe** [comment]
Hey

I haven't had the time to port the uic tool yet, but someone already did it https://github.com/stephenlyu/goqtuic

Please leave this issue open, as a reminder for me.

**Comment by tnajor** [comment]
> Hey
> 
> I haven't had the time to port the uic tool yet, but someone already did it https://github.com/stephenlyu/goqtuic
> 
> Please leave this issue open, as a reminder for me.

aha, thx!

**Comment by TCB13** [comment]
@tnajor thanks for the tip, however I'm not having much luck with it... https://github.com/stephenlyu/goqtuic/issues/5

**Comment by therecipe** [comment]
@tnajor @TCB13 

I recently added support for the uic with https://github.com/therecipe/qt/commit/01e05b1d5c9df871fabae4842127189431e526ac

Maybe checkout the difference between these examples https://github.com/therecipe/qt/tree/master/internal/examples/uitools/calculator
and
https://github.com/therecipe/qt/tree/master/internal/examples/uitools/calculator_manual

for this to work you will only need to update the binding and then place your `*.ui` files inside a folder called `ui`, then the binding (qtrcc/qtdeploy) should generate all the `widget.FindChild(` functions for you automatically or directly generate go code. You can use the `-uic` flag to switch between the embedding and go code generation mode.

**Comment by JasonWeise** [comment]
Hi

When using the built in support for ui files as per above I am recieiving the following error on a basic MainWindow:
```
ui/uic_mainwindow.go:58:2: undefinedL core.QMetaObject_ConnectSlotsByName
```

If I move the .ui file into another project and manually run 
```
goqtuic
```
from the terminal then the generated mainwindow_ui.go file displays fine.

The "undefined" error above is shown as soon as I run qtdeploy.

Looking for some guidance on why this might be happening with the therecipe version.

Thanks in advance.

**Comment by therecipe** [comment]
@JasonWeise 
Does line 58 just contain `core.QMetaObject_ConnectSlotsByName(w)`?
Which Qt version did you use?
Also does `qtdeploy -uic=false` work?

**Comment by JasonWeise** [comment]
thanks, I tried
```
qtdeploy -uic=false
```
and it worked.
if I just do `qtdeploy` then then line 58 does only contain:
core.QMetaObject_ConnectSlotsByName(w)

I am using QT 5.13.0-RC now (I was using Beta-4 when I first logged this above)
5.13.0-RC is also causing the same error on line 58.

What is the impact of ```-uic=false```

**Comment by therecipe** [comment]
>What is the impact of -uic=false
It disables the transpilation from the uic generate c++ code to go code, and instead embeds the `*.ui` file and then uses the `FindChildren` function to make the widgets available.

Can you reproduce the issue with https://github.com/therecipe/qt/tree/master/internal/examples/uitools/calculator?

**Comment by JasonWeise** [comment]
Quick update @therecipe 

I have tried a simple qtdeploy with the calculator demo and receive the same error (but this time on line 138)
```
ui/uic_calculatorform.go:138:2: undefined: core.core.QMetaObject_ConnectSlotsByName(w)
```

I have noticed too if I use -uic=false then you cannot use docker to build wasm
```
qtdeploy -uic=false -docker build wasm 
```
as it just errors with:
```
ui/uic_mainwindow.go:31:66: undefined: uitools.NewQUiLoader
```
if I just do a normal ```qtdeploy -uic=false``` it works fine on the desktop.

**Comment by therecipe** [comment]
@JasonWeise Thanks

How do you run qtdeploy? Do you use the `fast` flag or something?

I'm on macOS myself, so `QT_API=5.12.0 QT_VERSION=5.13.0 qtdeploy test desktop` should work for the calculator example.

Also does the uic_calculatorform.go file really contain `core.core.` or did that just happen during copy & past?

>I have noticed too if I use -uic=false then you cannot use docker to build wasm

Yeah, the web assembly version somehow does not support the uitools module atm, so you will need to use the (default) uic version to make it work with wasm/js targets.

Btw, I added images with thread support for the js/wasm targets as well, they are called `wasm_thread` and `js_thread`. You can use them like the regular wasm/js targets with `qtdeploy -docker build wasm_thread` (or `js_thread`)

Also maybe try to export `QT_NOT_CACHED=true` as well, it could be that this is a caching issue or something.
So maybe `QT_NOT_CACHED=true QT_API=5.12.0 QT_VERSION=5.13.0 qtdeploy test desktop` does work for you.

**Comment by JasonWeise** [comment]
Hey @therecipe 

SUCCESS!
Simplest of things and a good one for anyone else that comes across it.

I originally exported QT_API=5.12.0 at the terminal to get qtsetup to complete without an error (due to the current index issue with 5.13.0 from my other ticket)

What I didn't do was either put it in my .bash_profile OR re-export this at the terminal before qtdeploy (I presumed it would only be required to do qtsetup with 5.13.0.)

Once I did the following it build without error.
```
export QT_API=5.12.0
qtdeploy test desktop
```

Good example to be prepared for having to stray off the path when trying the bleeding edge :/

Thanks for all your help.

NOTE: core.core was due to my very poor copy / paste skills ;)
